src/UBAFeeCalculator/UBAFeeSpokeCalculator.ts

```diff
@@ -70,34 +81,46 @@ export default class UBAFeeSpokeCalculator {
     // If the last validated running balance is undefined, we need to compute the running balance from scratch
     // This is the case when the UBA Fee Calculator is first initialized or run on a range
     // that we haven't computed the running balance for yet
-    const historicalResult = this.recentRequestFlow.slice(startIdx, endIdx).reduce((acc, flow) => {
-      let resultantValue = acc;
-      if (isUbaInflow(flow)) {
-        resultantValue = acc.add(toBN(flow.amount));
-      } else if (isUbaOutflow(flow)) {
-        resultantValue = acc.sub(toBN(flow.amount));
+    const historicalResult: RunningBalanceUnion = this.recentRequestFlow.slice(startIdx, endIdx).reduce(
+      (acc, flow) => {
+        const resultant: RunningBalanceUnion = { ...acc };
```

Comment:
I was curious about the unpacking of ...acc here, since it's unconditionally overwritten below. Is it just to appease tsc and immediately infer the right type, or is there some other benefit?

Reply:
Modified here: https://github.com/across-protocol/sdk-v2/pull/212/commits/cf3bf9b7c89ff882322192fe973a16428a6ff703